core-bundle/src/DataContainer/Palette.php

```diff
@@ -0,0 +1,327 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Contao\CoreBundle\DataContainer;
+
+use Contao\StringUtil;
+
+class Palette implements \Stringable
+{
+    final public const POSITION_BEFORE = 'before';
+
+    final public const POSITION_AFTER = 'after';
+
+    final public const POSITION_PREPEND = 'prepend';
+
+    final public const POSITION_APPEND = 'append';
+
+    /**
+     * @param array<string|int, array{fields: array<string>, hide: bool}> $palette
+     */
+    private array $config;
+
+    public function __construct(string $palette = '')
+    {
+        $this->config = $this->explode($palette);
+    }
+
+    public function __toString(): string
+    {
+        return $this->toString();
+    }
+
+    public function toString(): string
+    {
+        return $this->implode();
```

Comment:
do we need a separate `implode()` method? Maybe we can just rename `implode` to `toString`?

Reply:
Done in https://github.com/contao/contao/pull/8302/commits/18a4b9490f701d34c1910c02439b3ffe5b4b6d1c